src/cryptoadvance/specter/templates/includes/helpers.jinja

```diff
@@ -1,26 +1,189 @@
 <!-- This file contains javascript helper functions that also can use jinja -->
 <script type="text/javascript">
 
-    // Formats an unitLabel (e.g. from a tx output) to a useful unit label
-    function formatUnitLabel(unitLabel, convertToSat){
-        var newLabel = unitLabel;
+
+
+    /*
+        Formats the btc amount as sats. 
+        Args:
+            value (Union[float, str]): (in BTC) Will convert string to float. 
+            enableDigitFormatting (bool, optional): Will group the Satoshis into blocks of 3,
+                e.g. 3 123 456, and color the blocks. Defaults to false.
+        Returns:
+            str: The formatted btc amount as html code.
+        */
+    function internalFormatBtcAmountAsSats(
+        value,
+        enableDigitFormatting=false,
+    ) {
+        const locale = 'en-US';
+        if (isNaN(parseFloat(value))){
+            return parseFloat(value)        
+        }
+        var s = (parseFloat(value) * 1e8).toLocaleString(locale, { 
+            minimumFractionDigits: 0, 
+            maximumFractionDigits: 0 
+        });         
+
+        let thousandsSeparator = Number(1000).toLocaleString(locale).charAt(1);
+        let decimalSeparator = Number(1.1).toLocaleString(locale).charAt(1);
+        
+        // combine the thousandsSeparator with the left number to an array
+        var array = [];
+        for (var i in s){
+            var letter = s[i];
+            if (letter == thousandsSeparator){
+                array[array.length-1] += letter;
```

Comment:
```suggestion
                array[array.length-1] += letter; // Using i would result in offset by 1 mistake
```

Reply:
Could you elaborate this comment? I am unsure what it is trying to say.

Reply:
I am generally trying with these helper functions to make them as simple as possible to improve maintainability and collaboration with other contributors, and here I initially thought I could just replace `[array.length-1] `with `i`, but since `i` is here stemming from the string `s`, `i` as index is off by one (actually only for the second thousand seperator). If it confuses even more, we can drop the comment.

Reply:
Good point. I added comments in [1ac96e6](https://github.com/cryptoadvance/specter-desktop/pull/1901/commits/1ac96e6473ac762c65c65ddb9d98e304715002ea)